dbtmetabase/parsers/dbt_manifest.py

```diff
@@ -201,12 +203,31 @@ def _read_model(
                 # would return the ref() written in the test, but if the model has an alias, that's not enough.
                 # It is better to use child['depends_on']['nodes'] and exclude the current model
 
-                depends_on_ids = set(child["depends_on"][model_type])
-                depends_on_ids.discard(model["unique_id"])
-                if not depends_on_ids:
+                # From experience, nodes contains at most two tables: the referenced model and the current model.
+                # Note, sometimes only the referenced model is returned.
+                depends_on_nodes = list(child["depends_on"][model_type])
+                if len(depends_on_nodes) > 2:
+                    logger().warning(
+                        "Expected at most two nodes, got %d {} nodes, skipping %s {}",
+                        len(depends_on_nodes),
+                        unique_id,
+                    )
+                    continue
+
+                # Remove the current model from the list. Note, remove() only removes the first occurrence. This ensures
+                # the logic also works for self referencing models.
+                if len(depends_on_nodes) == 2 and unique_id in depends_on_nodes:
```

Comment:
Do you need the `len == 2` condition here? You only care whether it's in there (and hence removing won't cause a KeyError), you're checking the length after that removal below for the ultimate return condition anyway.

Reply:
Yes I would say so. What if the nodes only contains the self referenced model, i.e., ‘len == 1’? In that case, we shouldn't remove it from the list.